<launch>
  <!-- Listen to messages from joysicks -->
  <!-- <node pkg="joy" name="joy_node" type="joy_node"/> -->

  <!-- Launch a map from the maps folder-->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/building_31.yaml"/> -->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/ucsd_atkinson_1f_full_tuned_edit.yaml"/>-->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/ucsd_atkinson_6f_icra2019_fx_net.yaml"/> -->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/grid_world_2.yaml"/> -->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/grid_world_2_edit.yaml"/> -->

  <!-- <arg name="map" default="$(find racecar_simulator)/maps/grid_world_2_small.yaml"/> -->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/intel.yaml"/> -->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/grid_world_2_0_06_small.yaml"/> -->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/grid_world_2_0_06.yaml"/> -->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/unknown.yaml"/> -->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/unknown2.yaml"/> -->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/grid_world_2_train2.yaml"/> -->

  <arg name="map" default="$(find racecar_simulator)/maps/mymap2.yaml"/>
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/mymap.yaml"/> -->
  <!-- <arg name="map" default="$(find racecar_simulator)/maps/narrowPassage.yaml"/> -->


  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map)"/>
  
  <!-- Launch the racecar model -->
  <include file="$(find racecar_simulator)/launch/racecar_model.launch"/>

  <!-- Begin the simulator with the parameters from params.yaml -->
  <node pkg="racecar_simulator" name="racecar_simulator" type="simulate" output="screen">
    <rosparam command="load" file="$(find racecar_simulator)/params.yaml"/>
  </node>

  <include file="$(find amcl)/examples/amcl_diff.launch" />
  
  <node pkg="move_base" type="move_base" respawn="true" name="move_base" output="screen">
  <!-- <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" launch-prefix="xterm -e gdb "> -->
    <rosparam file="$(find racecar_simulator)/params/costmap_common_config.yaml" command="load" ns="global_costmap" /> 
    <rosparam file="$(find racecar_simulator)/params/costmap_common_config.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find racecar_simulator)/params/costmap_common_config.yaml" command="load" ns="collision_costmap"/>
    <rosparam file="$(find racecar_simulator)/params/local_costmap_param.yaml" command="load" />
    <rosparam file="$(find racecar_simulator)/params/global_costmap_param.yaml" command="load" /> 
    <rosparam file="$(find mpnet_plan)/params/collision_costmap_param.yaml" command="load" /> 
    <param name="NavfnROS/allow_unknown" value="false"/>
    <param name="global_costmap/obstacle_costmap/track_unknown_space" value="true"/>
    <param name="base_local_planner" value="mpnet_local_planner/MpnetLocalPlanner" />
    <param name="controller_frequency" value="5.0"/>
    <!-- <param name="controller_patience" value="15.0"/> -->

    <rosparam file="$(find mpnet_plan)/params/local_planner.yaml" command="load" />
  </node>

  <!-- <node pkg="mpnet_plan" type="controller_node" respawn="true" name="controller_node" output="screen"/> -->
  <node pkg="mpnet_plan" type="pure_pursuit_controller.py" respawn="true" name="pure_pursuit" output="screen"/>


  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find  mpnet_plan)/launch/mpnet_launch.rviz" />
  <!-- <node type="rviz" name="rviz" pkg="rviz"/> -->


</launch>
